src/controllers/factorycontroller.js

```diff
@@ -32,6 +32,18 @@ class FactoryController extends OrderController
             }
         );
 
+        //form submission, adds an extra truck after cost validation
+        $("#buy-truck").submit(() => {
```

Comment:
`click`, not `submit`. It is a button, not a form. Also, the comment above this line needs some updating :).